### ISSUE TITLE
Don't extend from EmberObject

### ISSUE THREAD

**Issue by nlfurniss** [hjdivad/ember-m3#293]
The Megamorpgic model [extends EmberObject](https://github.com/hjdivad/ember-m3/blob/d837b61f6f74978a83e553033f1e4e16e6511017/addon/model.js#L111), but that behavior is [deprecated](https://github.com/hjdivad/ember-m3/blob/d837b61f6f74978a83e553033f1e4e16e6511017/addon/model.js#L111) and is being removed in 3.9.

The deprecation guide mentions the need to `write your own constructor function`; I would tackle this but I'm unsure what that constructor would need to replicate from EO.

**Comment by runspired** [comment]
We would just need to cleanup this: https://github.com/hjdivad/ember-m3/pull/274

**Comment by nlfurniss** [comment]
Ahh right, I remember that PR now

**Comment by hjdivad** [comment]
fixed by #276

**Comment by runspired** [comment]
@hjdivad the deprecations were resolved bu #276 but we still extend EmberObject when we probably shouldn't for MegamorphicModel. Should we open a new ticket for eliminating EmberObject once custom record classes lands?

**Comment by hjdivad** [comment]
@runspired sgtm

**Comment by runspired** [comment]
Created https://github.com/hjdivad/ember-m3/issues/362
